vck-rqes/src/commonMain/kotlin/at/asitplus/wallet/lib/rqes/RqesOidcVerifier.kt

```diff
@@ -0,0 +1,70 @@
+package at.asitplus.wallet.lib.rqes
+
+import at.asitplus.dif.DifInputDescriptor
+import at.asitplus.dif.InputDescriptor
+import at.asitplus.openid.AuthenticationRequestParameters
+import at.asitplus.openid.AuthenticationResponseParameters
+import at.asitplus.rqes.QesInputDescriptor
+import at.asitplus.rqes.collection_entries.TransactionData
+import at.asitplus.wallet.lib.agent.DefaultCryptoService
+import at.asitplus.wallet.lib.agent.DefaultVerifierCryptoService
+import at.asitplus.wallet.lib.agent.EphemeralKeyWithoutCert
+import at.asitplus.wallet.lib.agent.KeyMaterial
+import at.asitplus.wallet.lib.agent.Verifier
+import at.asitplus.wallet.lib.agent.VerifierAgent
+import at.asitplus.wallet.lib.data.vckJsonSerializer
+import at.asitplus.wallet.lib.jws.DefaultJwsService
+import at.asitplus.wallet.lib.jws.DefaultVerifierJwsService
+import at.asitplus.wallet.lib.jws.JwsService
+import at.asitplus.wallet.lib.jws.VerifierJwsService
+import at.asitplus.wallet.lib.oidc.OidcSiopVerifier
+import at.asitplus.wallet.lib.oidvci.DefaultMapStore
+import at.asitplus.wallet.lib.oidvci.DefaultNonceService
+import at.asitplus.wallet.lib.oidvci.MapStore
+import at.asitplus.wallet.lib.oidvci.NonceService
+import io.ktor.util.reflect.*
+import kotlinx.datetime.Clock
+
+/**
+ * Verifier with access to [TransactionData] class can now generate requests containing [TransactionData]
+ */
+class RqesOidcVerifier(
```

Comment:
Maybe even override the wohle `createAuthnRequest()`, or even better provide an extension point in `OidcSiopVerifier`, so that we can move all those `rqesParameters` into this class here?